actions/command.js

```diff
@@ -97,24 +97,26 @@ export function executeCommand(message, args) {
             dispatch(PostActions.resetEmbedVisibility());
         }
 
-        const parser = new AppCommandParser({dispatch, getState}, args.root_id);
-        if (parser.isAppCommand(msg)) {
-            try {
-                const call = await parser.composeCallFromCommandString(msg);
-                if (!call) {
-                    return {error: new Error('Error composing command submission')};
+        if (shouldProcessApps(state)) {
```

Comment:
To avoid adding the arrow code (https://blog.codinghorror.com/flattening-arrow-code/), why not add a guard at the start of each root method smth like : `if (!shouldProcessApps(state)) { return;}`

Reply:
We don't want to return. We want to skip this piece of code.

Reply:
We could extract that part to a function, but we should notify whether we have to return or not, and what. Something like:
```
if (shouldProcessApps(state)) {
    const data = processAppCommand()
    if (data) {return data};
}
```

Reply:
There maybe some other methods down below that could use the short-circuiting.

Reply:
The next part of the code is directly executing the command on the server as a normal command, so I cannot promote anything upwards. I don't see any other way to simplify this.